lib/gooddata/models/user_filters/user_filter_builder.rb

```diff
@@ -584,12 +628,12 @@ def self.normalize_filters(filters)
     # Removes MUFs from to_delete unless in user is in users_brick_input
     # if this does not happen, users that are about to be deleted by users_brick
     # would have all their filters removed now, which is not desirable
-    def self.sanitize_filters_to_delete(to_delete, users_brick_input, project_users)
+    def self.sanitize_filters_to_delete(to_delete, users_brick_input, user_profile_mapping)
       return [] unless users_brick_input && users_brick_input.any?
       user_profiles = users_brick_input.map do |user|
-        result = project_users.find { |u| u.login == user.with_indifferent_access['login'] }
+        result = user_profile_mapping[user.with_indifferent_access['login']]
```

Comment:
user_profile_mapping don't contains user of deleted filters, only input_filters.

Reply:
The delete MUF will be calculated from users_brick_input, to_delete and input_filters.